### DISCUSSION TITLE
using `expand` while `viewing` or `listing` needs further demonstration.

### DISCUSSION THREAD

**Discussion by Mr-M1M3** [pocketbase/pocketbase#782] [Q&A] [answered]
Hello, I cannot expand a relation field with the official js-sdk.

I have a collection named `post` and this collection has a required relation field named `author`.
Anyone can view or list this collection. Other rules are,
Create Rule: 
```@request.user.id != ""```
Update and Delete rule of this collection is: 
```author = @request.user.id```

#
On the client side, I tried the snipped below to `view` a post with `expand` property set to `author`

```javascript

async function viewPost(){
    const post = await client.records.getOne('posts', 'record_id' , {
        expand: "author"
    });
    console.log(post);
}

viewPost();
```
The above snippet does work but, `@expand` property is an empty object. How do I fix that?

**Comment by ganigeorgiev** [comment, marked as answer]
If `author` is a `user` type field, then you can't expand the related user data.
Instead you can change it to a `relation` type field pointing to the `profilles` collection and associate the profile id instead of the user id.

In the access rules you can access the profile id by `@request.user.profile.id`.

This `user`/`profiles` relation confusion will be fixed very soon with #376 in v0.8.

**Reply by Mr-M1M3** [reply]
@ganigeorgiev I got it!! I thought I set `author` field to a `relation` type. Turned out, I set it to a user type field 😅😅😅
